### DISCUSSION TITLE
[suggestion] Warn user if PDF is interactive form

### DISCUSSION THREAD

**Discussion by Shohreh** [sumatrapdfreader/sumatrapdf#4609] [Ideas]
(I haven't found how to check if the suggestion has already been made in the forum)

Hello,

Since SumatraPDF doesn't yet support forms, I suggest at least warning the user when opening such files like Foxit Reader does.

I was sent a form PDF recently. Since Sumatra just displayed it with no warning, I thought it was just bad design and painfully edited it in LibreOffice Draw… before being told by the sender that I could have filled it directly :-/

Cheers,

![image](https://github.com/user-attachments/assets/2390a5f5-2965-4ab9-bae8-ce3566c103c1)

**Comment by edersilvago** [comment]
When will we have support for filling out forms?
![Screenshot_1](https://github.com/user-attachments/assets/5107a12c-21f6-44ec-8b1b-7bb383de2ed7)

**Comment by GitHubRulesOK** [comment]
This is not a simple topic there are actually several ways forms are constructed.

SumatraPDF is based on MuPDF which does have limited ISO Standard AcroForm filling Abilities (for most basic JetForm methods). but not enabled as problematic to add a user interface for all types (MuPDF uses floating boxes).

Any forms with Adobe XFA (usually Government forms) are to be untrusted for reliable behaviour **outside of Adobe server linked products** e.g. MANDATE using Acrobat Reader, since they are non conformant with ISO standard and can thus need a warning like "This is an XFA" inside a single page PDF.

In addition Adobe Acrobat has complex "on the fly" form field **editor** abilities for lines and check boxes.
so this "Form" has NO Form areas but Acrobat Reader will ADD them. we can confirm that in that case there are no form fields
![image](https://github.com/user-attachments/assets/f710c1a1-83b2-4938-bbb5-cc0bf13d6b3a)
![image](https://github.com/user-attachments/assets/5efcaa08-4e3e-4f2f-b9e5-e5a955a83e1c)
